### PR TITLE
feat(chat): add backend for the cross-check chat

### PR DESCRIPTION
**Pull request by igorlyatskiy**
#### 🧑‍⚖️ Pull Request Naming Convention

 - Title should follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)
 - Do not put issue id in title
 - Do not put WIP in title. Use [Draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/) functionality
 - Consider to add `area:*` label(s)

 * [x] I followed naming convention rules

---

#### 🤔 This is a ...
- [x] New feature
- [ ] Bug fix
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Test Case
- [ ] Documentation update
- [ ] Other

#### 🔗 Related issue link

_Describe the source of requirement, like related issue link._

#### 💡 Background and solution 

_Describe the big picture of your changes here_

#### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Database migration is added or not needed
- [x] Documentation is updated/provided or not needed
- [x] Changes are tested locally

### CODE REVIEW

File: nestjs/src/courses/crossCheck/chat/chat.service.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+  public async getTaskSolutionChat(pairId: number) {
+    const taskSolution = await this.taskSolutionResultRepository.findOne({
+      where: {
+        id: pairId,
+      },
+      relations: ['chat'],
+    });
+
+    if (taskSolution.chat == null) {
```
Comment (KostyaDedulya) [inline nestjs/src/courses/crossCheck/chat/chat.service.ts:36]: !taskSolutuin.chat

File: nestjs/src/courses/crossCheck/chat/chat.service.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    const history = taskSolution?.chat?.history || [];
+
+    return { chat: history };
+  }
+
+  private async getStudentAndCourseTask(courseId: number, githubId: string, courseTaskId: number) {
+    const { student, courseTask } = await this.crossCheckService.getStudentAndTask(courseId, githubId, courseTaskId);
+
+    if (student == null || courseTaskId == null) {
```
Comment (KostyaDedulya) [inline nestjs/src/courses/crossCheck/chat/chat.service.ts:48]: !student || !courseTaskId

File: nestjs/src/courses/crossCheck/chat/chat.service.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+import { CrossCheckService } from '../crossCheck.service';
+
+@Injectable()
+export class ChatService {
+  constructor(
+    @InjectRepository(TaskSolutionResult)
+    private readonly taskSolutionResultRepository: Repository<TaskSolutionResult>,
+    @InjectRepository(TaskSolutionChat)
+    private readonly taskSolutionResultChat: Repository<TaskSolutionChat>,
```
Comment (KostyaDedulya) [inline nestjs/src/courses/crossCheck/chat/chat.service.ts:16]: naming

File: nestjs/src/courses/crossCheck/chat/dto/messageDto.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+  @ApiProperty()
+  date: string;
+
+  @ApiProperty()
+  text: string;
+
+  @ApiProperty()
+  isRead: boolean;
+}
```
Comment (KostyaDedulya) [inline nestjs/src/courses/crossCheck/chat/dto/messageDto.ts:15]: validator

File: server/src/models/taskSolutionChat.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  updatedDate: number;
+
+  @Column()
+  taskSolutionResultId: number;
+
+  @Column({
+    type: 'jsonb',
+    default: []
+  })
+  history: Message[];
```
Comment (KostyaDedulya) [inline server/src/models/taskSolutionChat.ts:33]: constructor(history = []) {}

File: server/src/services/crossCheck.service.ts [hunk 6]
```diff
@@ -126,27 +128,37 @@ export class CrossCheckService {
...
         ...data,
       });
+
+      const id = taskSolutionResult.identifiers[0]?.id;
+      if (id) {
+        const taskSolutionChatRepo = getRepository(TaskSolutionChat);
+
+        const taskSolutionChat = new TaskSolutionChat();
+        taskSolutionChat.history = [];
+        taskSolutionChat.taskSolutionResultId = id;
```
Comment (KostyaDedulya) [inline server/src/services/crossCheck.service.ts:145]: remove

**Review by KostyaDedulya — CHANGES_REQUESTED** [review]

**Review by kdziadziulia — APPROVED** [review]

**Review by ilyatskiy — APPROVED** [review]

**Pull request merged**